sw/c/demo/i2c_hat_id/i2c_hat_id.c

```diff
@@ -21,6 +30,13 @@ int main(void) {
   static uint8_t data[0x80u];
 
   uart_init(DEFAULT_UART);
+  rv_plic_init();
+
+  // Test IRQ
+  rv_plic_register_irq(fmt_threshold_irq, i2c_irq_test);
+  rv_plic_enable(fmt_threshold_irq);
+  DEV_WRITE(DEFAULT_I2C + 0x4,  0x1);
+  DEV_WRITE(DEFAULT_I2C + 0x24, 0x10000);
```

Comment:
Might be nice to put a few comments in here to say that this enables interrupt and set format threshold.